src/shared/watchedFiles.ts

```diff
@@ -106,10 +106,17 @@ export abstract class WatchedFiles<T> implements vscode.Disposable {
 
     /**
      * Get a specific item's data
-     * @param path Absolute path to template of interest
+     * @param path Absolute path to item of interest or a vscode.Uri to the item
      */
-    public getRegisteredItem(path: string): WatchedItem<T> | undefined {
-        const normalizedPath = pathutils.normalize(path)
+    public getRegisteredItem(path: string | vscode.Uri): WatchedItem<T> | undefined {
+        let absolutePath: string
+        if (typeof path === 'string') {
+            absolutePath = path
+        } else {
```

Comment:
Seems like a half-measure. Is there a reason we wouldn't want to send a `vscode.Uri`?

Reply:
we use it a lot with constructed paths, so no reason to coerce them into `vscode.Uri`'s if that's not what we start with

Reply:
That actually makes me extremely "eh" on this change then. We typically use `vscode.Uri.fsPath` everywhere else...`path` vs. `fsPath` will show up in a lot of other places and is definitely not exclusive to here.

Reply:
why do we use fsPath everywhere though? I don't think the caller should have to know what field to pluck out

Reply:
`fsPath` is OS-specific: https://code.visualstudio.com/api/references/vscode-api#Uri.fsPath ; `path` would be fine for URLs. re: constructed paths, it's not hard to convert to a `vscode.Uri`...use `vscode.Uri.file()`.

Whatever we do, I think we should standardize with `addItemToRegistry`. It doesn't make a lot of sense that you add via URI and you get via string.

Reply:
A half-measure is ok if it's a monotonic improvement, no? I don't see any debt incurred by this PR and further changes are 2-way door.